### ISSUE TITLE
:xmldecl needs to use :attribute_quote context setting

### ISSUE THREAD

**Issue by konung** [ruby/rexml#8]
### History of the issue
_Moving https://github.com/ruby/ruby/pull/496#issuecomment-430522768 to this repo -> ruby/rexml_ which in turn is a copy of https://redmine.ruby-lang.org/issues/9367#note-10 

### The crux of the issue (original comment) 
from @bearmini

REXML uses double quotes for quoting attributes if :attribute_quote is specified as document's context like below:
```
doc = REXML::Document.new
doc.context[:attribute_quote] = :quote
```

This looks working well on all elements but has no effect for xml declaration (i.e. <?xml version= ... ?>) if it exists.

Even if I specify (({doc.context[:attribute_quote] = :quote})), I will get this:

```xml
<?xml version='1.0' encoding='UTF-8' standalone='true'?>
```

The expected result is:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="true"?>
```

### Confirming the problem and the rationale 
from @konung

I ran into the same issue. Since there seems to be a question of "why would you want to do that?", based on the discussion on Rubymine, I'd like to add a comment.
I came across this when dealing with XML generated for consumption by a Java (Jetty) based service (used in ebXML exchange popular in Automotive industry).

   

1. It complains the file is malformed and won't parse it if the XML declaration is not quoted, like the rest of the document.
2. It also doesn't want to parse single quoted attributes in XML file. ( Specifying doc = REXML::Document.new(source); doc.context[:attribute_quote] = :quote -> works, just not for the xml declaration!)
3. Finally it doesn't like '\n' UNIX-style line endings, and wants Windows-style endings ( I sidestepped the issue by using File.write(src, str, crlf_newline: true) when generating final file. )

So the only remaining problem is 1 - right now I'm working around the issue by parsing the file before saving and replacing single quotes with double quotes in xml declaration. But I wanted to explain why this is a useful option, despite not being 100% true to the standard.

Thank you

### Nokogori
I tested the same xml in Nokogiri, and it seems to convert quotes in the xmldec to double quotes, like the rest of the document. 

While the standard supports both double and single quotes, double quotes seems to have more consistent support throughout different implementations. (IMHO)

**Comment by konung** [comment]
@kou Kouhei, no problem. 

I don't really know which particular XML parser the Java application is using - it's closed source and commercial product that's popular in Automotive industry - from a company called Axway. It designed to exchanged in the ebXML format, popular with car companies. 

But when I tried to test using single quotes XML it kept throwing very cryptic errors at me about malformed files, until I changed them all to double-quoted.
